src/htm/engine/RESTapi.cpp

```diff
@@ -47,7 +47,7 @@ RESTapi* RESTapi::getInstance() { return &rest; }
   while (rest.resource_.size() < ID_MAX) {  // limit the total number of generated resources
     unsigned int id_nbr = next_id++;
     if (id_nbr > ID_MAX)
-      id_nbr = 1; // allow integer wrap of the id without a "0000" value.
+      id_nbr = 1; // allow integer wrap of the id without using a "0000" value.
```

Comment:
OT: do we want to be using the "zero padded" numbers (as a text)? Wouldn't it cause problems parsing back to numbers? Why not returning "1", rather than "0001" ?

Reply:
First, the id is a string not a number.  The characters don't even need to be digits.  So there is never a reason to convert to a numeric value.

When the id value is returned in the body of the response the protocol appends a '\n' to the end of the string.  The client then needed to strip off the '\n' character to get the full id.  At first I was just grabbing the first 4 characters as the id so that is why I padded with 0's (and set the max to be '9999'.  Since then I changed the logic to just remove the last character which allows the id to be any length.  So yes we could return just "1" but it is entirely arbitrary and so I just left it as it was with the 4 characters.

Thanks for the reviews.